admin/tools/provision/cloudmanager.py

```diff
@@ -299,7 +305,8 @@ def _build_instance_name(self, instance_id):
         :param instance_id: instance id
         :return:            instance name
         """
-        if isinstance(instance_id, unicode):
+        # TODO: not sure if the logic here is doing what it is supposed to do
+        if isinstance(instance_id, basestring):
```

Comment:
To me it seems that real intent here is to remove "bad" characters from `instance_id`, but I'm not sure that conversion to bytes is wanted. @fjammes should probably comment on this, I'll ask him to review this code.